ui/src/app/pages/analysis/interactive-notebook.spec.tsx

```diff
@@ -0,0 +1,87 @@
+import { MemoryRouter, Route } from 'react-router-dom';
+
+import { AppStatus, WorkspaceAccessLevel } from 'generated/fetch';
+
+import { screen } from '@testing-library/dom';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import * as swaggerClients from 'app/services/swagger-fetch-clients';
+import { currentWorkspaceStore } from 'app/utils/navigation';
+import { userAppsStore } from 'app/utils/stores';
+
+import { createListAppsRStudioResponse } from 'testing/stubs/apps-api-stub';
+import { workspaceStubs } from 'testing/stubs/workspaces';
+
+import { InteractiveNotebook } from './interactive-notebook';
+
+let mockNotebooksApi;
+
+beforeEach(async () => {
+  currentWorkspaceStore.next({
+    ...workspaceStubs[0],
+    accessLevel: WorkspaceAccessLevel.WRITER,
+  });
+
+  jest.mock('app/services/swagger-fetch-clients');
+  mockNotebooksApi = jest.spyOn(swaggerClients, 'notebooksApi');
+});
+
+const setup = (mockOverrides) => {
+  mockNotebooksApi.mockImplementation(() => ({
+    ...mockOverrides,
+  }));
+};
+
+const renderInteractiveNotebook = (pathParameters) =>
+  render(
+    <MemoryRouter
+      initialEntries={[
+        '/workspaces/sampleNameSpace/sampleWorkspace/notebooks/preview/example.Rmd',
+      ]}
+    >
+      <Route path='/workspaces/:ns/:wsid/notebooks/preview/:nbName'>
+        <InteractiveNotebook hideSpinner={() => {}} match={pathParameters} />
+      </Route>
+    </MemoryRouter>
+  );
+
+test('Edit Rmd file with running RStudio', async () => {
+  setup({
+    getNotebookLockingMetadata: () => Promise.resolve({}),
+  });
+  const rStudioApp = createListAppsRStudioResponse({
+    status: AppStatus.RUNNING,
+  });
+  userAppsStore.set({
+    userApps: [rStudioApp],
+  });
+  const pathParameters = { params: { nbName: 'test.Rmd' } };
+  const spyWindowOpen = jest.spyOn(window, 'open');
+  spyWindowOpen.mockImplementation(jest.fn(() => window));
+  renderInteractiveNotebook(pathParameters);
+  const editButton = screen.getByTitle('Edit');
+  fireEvent.click(editButton);
+  await waitFor(() => {
+    expect(spyWindowOpen).toHaveBeenCalledTimes(1);
+    expect(spyWindowOpen).toHaveBeenCalledWith(
+      rStudioApp.proxyUrls['rstudio-service'],
+      '_blank'
+    );
+  });
+});
+
+test('Edit Rmd file with no running RStudio', async () => {
```

Comment:
I'm not sure if this test is useful or not. I would love thoughts here.

I say that it is not helpful, because in terms of this component, nothing changes when you click edit. We add the RStudio icon in the SidebarActiveIconStore, but checking that that value ends up in the store feels like a implementation detail that does not need to be tested.

Eventually, I think that this condition might be more testable, because we could update the "Edit" button's text to reflect that an app is being created (or something of the sort), and we could test for that. Until then, I am ensuring window.open is not called (as it would be if there was a running RStudio app).

Reply:
> checking that that value ends up in the store feels like a implementation detail that does not need to be tested

I'd call this a side effect of the component, which is worth testing (and documenting) here.

This is where the `test("should X when Y"` naming convention can be helpful. What should happen when the app isn't running and you click edit?